pkgs/applications/misc/calibre/default.nix

```diff
@@ -134,20 +179,39 @@ stdenv.mkDerivation (finalAttrs: {
     ] ++ lib.optional (unrarSupport) unrardll
   );
 
+  env = {
+    POPPLER_INC_DIR = "${poppler_utils.dev}/include/poppler";
+    POPPLER_LIB_DIR = "${poppler_utils.out}/lib";
+    MAGICK_INC = "${imagemagick.dev}/include/ImageMagick";
+    MAGICK_LIB = "${imagemagick.out}/lib";
+    FC_INC_DIR = "${fontconfig.dev}/include/fontconfig";
+    FC_LIB_DIR = "${fontconfig.lib}/lib";
+    PODOFO_INC_DIR = "${podofo.dev}/include/podofo";
+    PODOFO_LIB_DIR = "${podofo.lib}/lib";
+    ISOCODE_ZIP = "${iso-codes-zip}";
+    ISOCODE_VERSION = iso-codes-zip.version;
+    CACERT = "${cacert}/etc/ssl/certs/ca-bundle.crt";
+    TRANSLATIONS = "${translations}";
+    # this is get from https://code.calibre-ebook.com/ua-popularity
+    # It's not reproducible, so we have to pack it in repo
+    # this file can be generated using `python fetch-ua.py`
+    UA_POPULARITY = "${./ua-popularity.txt}";
+  };
+
+  buildPhase = ''
+    mkdir temp
+    cp "${cacert}/etc/ssl/certs/ca-bundle.crt" resources/mozilla-ca-certs.pem
+    ${python3Packages.python.pythonOnBuildForHost.interpreter} setup.py bootstrap \
+      --path-to-mathjax=${mathjax} \
+      --path-to-liberation_fonts=${liberation_fonts} \
+      --path-to-hyphenation=${hyphenation}
+    ${python3Packages.python.pythonOnBuildForHost.interpreter} setup.py man_pages
+  '';
+
   installPhase = ''
     runHook preInstall
 
     export HOME=$TMPDIR/fakehome
-    export POPPLER_INC_DIR=${poppler_utils.dev}/include/poppler
-    export POPPLER_LIB_DIR=${poppler_utils.out}/lib
-    export MAGICK_INC=${imagemagick.dev}/include/ImageMagick
-    export MAGICK_LIB=${imagemagick.out}/lib
-    export FC_INC_DIR=${fontconfig.dev}/include/fontconfig
-    export FC_LIB_DIR=${fontconfig.lib}/lib
-    export PODOFO_INC_DIR=${podofo.dev}/include/podofo
-    export PODOFO_LIB_DIR=${podofo.lib}/lib
-    export XDG_DATA_HOME=$out/share
-    export XDG_UTILS_INSTALL_MODE="user"
```

Comment:
`XDG_*` variables seem to be missing in the new version. Are they not necessary anymore?

Reply:
I rendered this obsolete in https://github.com/kovidgoyal/calibre/pull/988 back in 2019.

Reply:
My fault. After some test I found `$out/share/{applications,icons,mime}` would disappear without these environment variables. I have added them back.

Reply:
But why are they needed? calibre will detect that it is run in packaging mode and set it to the correct datadirs without you messing around with INSTALL_MODE...